mopidy/local/playlists.py

```diff
@@ -23,18 +22,27 @@ def __init__(self, *args, **kwargs):
         self.refresh()
 
     def create(self, name):
-        name = formatting.slugify(name)
-        uri = 'local:playlist:%s.m3u' % name
-        playlist = Playlist(uri=uri, name=name)
-        return self.save(playlist)
+        playlist = self._save_m3u(Playlist(name=name))
+        old_playlist = self.lookup(playlist.uri)
+        if old_playlist is not None:
+            index = self._playlists.index(old_playlist)
+            self._playlists[index] = playlist
+        else:
+            self._playlists.append(playlist)
+        logger.info('Created playlist %s', playlist.uri)
+        return playlist
 
     def delete(self, uri):
         playlist = self.lookup(uri)
         if not playlist:
+            logger.warn('Trying to delete unknown playlist %s', uri)
             return
-
+        path = local_playlist_uri_to_path(uri, self._playlists_dir)
+        if os.path.exists(path):
+            os.remove(path)
```

Comment:
I'm a little undecided about this, giving your instructions about `OSError` and handling it with logging ;-)
The old code read:
``
       file_path = self._m3u_uri_to_path(uri)
       if os.path.exists(file_path):
           os.remove(file_path)
```
else "nada", so I tried to keep close to that, but do some logging in the `else` branch (I spent a whole evening trying to figure out what was actually wrong with local playlists, and this would have definitely helped),